CHANGELOG.md

```diff
@@ -7,6 +7,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Removed
+
+- BREAKING CHANGE: Removed deprecated `var.defaults`
+- BREAKING CHANGE: Removed deprecated `var.private`, use `var.visibility` instead
+
```

Comment:
no need to remove them already unless we refactor the complete module.

Reply:
The problem is that the module is barely maintainable due to the complexity of defaults. I'd still like to merge this PR, migrate to the new test suite as a next step and then start adding more features requested by the community. WDYT?

Reply:
new feature do not need to be added to the complex defaults.. this module is on the list of complete refactor. There is no need to add breaking changes now.